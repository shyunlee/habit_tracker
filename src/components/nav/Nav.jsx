import React, { PureComponent } from 'react';

class Nav extends PureComponent {

  render() {
    return (
      <nav className="nav">
        <div className="nav-bar">
          <span className="nav-logo">
            <i className="fas fa-leaf"></i>
          </span>
          <span className="nav-title">My Habits</span>
          <span className="nav-count" data-testid='total-count'>{this.props.totalCount}</span>
        </div>
      </nav>
    );
  }
}

export default Nav;