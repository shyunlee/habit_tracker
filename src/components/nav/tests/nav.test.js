import React from 'react';
import renderer from 'react-test-renderer';
import Nav from '../Nav';

describe('Nav', () => {
  it('renders', () => {
    const component = renderer.create(<Nav totalCount={4} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
