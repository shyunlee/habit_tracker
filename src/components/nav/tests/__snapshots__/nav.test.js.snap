// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders 1`] = `
<nav
  className="nav"
>
  <div
    className="nav-bar"
  >
    <span
      className="nav-logo"
    >
      <i
        className="fas fa-leaf"
      />
    </span>
    <span
      className="nav-title"
    >
      Habit Tracker
    </span>
    <span
      className="nav-count"
      data-testid="total-count"
    >
      4
    </span>
  </div>
</nav>
`;
